api/v1alpha1/tinkerbell_type.go

```diff
@@ -0,0 +1,125 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=tinkerbell,scope=Namespaced,categories=tinkerbell,singular=tinkerbell,shortName=tb
+// +kubebuilder:storageversion
+
+// Tinkerbell represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
+// Tinkerbell operator watches for different resources such as deployment, services, serviceAccounts, etc. One of those
+// CRs is Tinkerbell which the operator will install the tink-stack based on its specs. Once the CR is deleted,
+// the operator will delete all tinkerbell resources.
+type Tinkerbell struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec describes the desired tinkerbell stack state.
+	Spec TinkerbellSpec `json:"spec,omitempty"`
+}
+
+// TinkerbellSpec specifies details of an Tinkerbell setup.
+type TinkerbellSpec struct {
+	// TinkerbellVersion is the tinkerbell crd version.
+	TinkerbellVersion string `json:"tinkerbellVersion"`
+
+	// ClusterDNS is the ip address of the cluster dns resolver.
+	ClusterDNS string `json:"clusterDNS"`
+
+	// OverwriteRegistry is the registry to use for all images. If this field is set, all tink service deployment images
+	// will be prefixed with this value. For example if the value here was set to docker.io, then boots image will be
+	// docker.io/tinkerbell/boots.
+	OverwriteRegistry string `json:"overwriteRegistry"`
+
+	// DockerPullConfig the secret name containing the docker auth config which should exist in the same namespace where
+	// the operator is deployed(typically tinkerbell)
+	DockerPullComfig string `json:"dockerPullComfig"`
+
+	// Boots contains all the information and spec about boots.
+	Boots BootsSpec `json:"boots"`
+	// Hegel contains all the information and spec about boots.
+	Hegel HegelSpec `json:"hegel"`
+	// Rufio contains all the information and spec about rufio.
+	Rufio RufioSpec `json:"rufio"`
+	// Tink contains all the information and spec about tink.
+	Tink TinkSpec `json:"tink"`
```

Comment:
```suggestion
	// Boots contains all the information and spec about Boots.
	Boots BootsSpec `json:"boots"`
	
	// Hegel contains all the information and spec about Hegel.
	Hegel HegelSpec `json:"hegel"`
	
	// Rufio contains all the information and spec about Rufio.
	Rufio RufioSpec `json:"rufio"`
	
	// Tink contains all the information and spec about Tink.
	Tink TinkSpec `json:"tink"`
```

Do we need to append `Spec` to all these? It feels a little forced toward the spec and status guidelines but its different to that.

Reply:
Not quite sure if I get this comment. Can you please elaborate a bit more.

Reply:
Sorry, I was referencing the types. For example, can `TinkSpec` just be `Tink`?